consensus/misc/eip4844/eip4844.go

```diff
@@ -62,38 +129,41 @@ func VerifyEIP4844Header(config *params.ChainConfig, parent, header *types.Heade
 // CalcExcessBlobGas calculates the excess blob gas after applying the set of
 // blobs on top of the excess blob gas.
 func CalcExcessBlobGas(config *params.ChainConfig, parent *types.Header, headTimestamp uint64) uint64 {
-	var (
-		parentExcessBlobGas uint64
-		parentBlobGasUsed   uint64
-	)
+	isOsaka := config.IsOsaka(config.LondonBlock, headTimestamp)
+	bcfg := latestBlobConfig(config, headTimestamp)
+	return calcExcessBlobGas(isOsaka, bcfg, parent)
+}
+
+func calcExcessBlobGas(isOsaka bool, bcfg *BlobConfig, parent *types.Header) uint64 {
+	var parentExcessBlobGas, parentBlobGasUsed uint64
 	if parent.ExcessBlobGas != nil {
 		parentExcessBlobGas = *parent.ExcessBlobGas
 		parentBlobGasUsed = *parent.BlobGasUsed
 	}
+
 	var (
 		excessBlobGas = parentExcessBlobGas + parentBlobGasUsed
-		target        = targetBlobsPerBlock(config, headTimestamp)
-		targetGas     = uint64(target) * params.BlobTxBlobGasPerBlob
+		targetGas     = uint64(bcfg.Target) * params.BlobTxBlobGasPerBlob
 	)
 	if excessBlobGas < targetGas {
 		return 0
 	}
-	if !config.IsOsaka(config.LondonBlock, headTimestamp) {
-		// Pre-Osaka, we use the formula defined by EIP-4844.
-		return excessBlobGas - targetGas
-	}
 
-	// EIP-7918 (post-Osaka) introduces a different formula for computing excess.
-	var (
-		baseCost     = big.NewInt(params.BlobBaseCost)
-		reservePrice = baseCost.Mul(baseCost, parent.BaseFee)
-		blobPrice    = calcBlobPrice(config, parent)
-	)
-	if reservePrice.Cmp(blobPrice) > 0 {
-		max := MaxBlobsPerBlock(config, headTimestamp)
-		scaledExcess := parentBlobGasUsed * uint64(max-target) / uint64(max)
-		return parentExcessBlobGas + scaledExcess
+	// EIP-7918 (post-Osaka) introduces a different formula for computing excess,
+	// in cases where the price is lower than a 'reserve price'.
+	if isOsaka {
+		var (
+			baseCost     = big.NewInt(params.BlobBaseCost)
+			reservePrice = baseCost.Mul(baseCost, parent.BaseFee)
+			blobPrice    = bcfg.blobPrice(excessBlobGas)
```

Comment:
```suggestion
			blobPrice    = bcfg.blobPrice(parentExcessBlobGas)
```

I believe this was the issue Marius found and fixed here: https://github.com/fjl/go-ethereum/compare/eip4844-transition-updatefraction...MariusVanDerWijden:go-ethereum:eip4844-transition-updatefraction?expand=1